Testing/Unit/PowerShell/Connection/Disconnect-SCuBATenant.Tests.ps1

```diff
@@ -0,0 +1,49 @@
+Import-Module (Join-Path -Path $PSScriptRoot -ChildPath "../../../../PowerShell/ScubaGear/Modules/Connection/Connection.psm1") -Function 'Disconnect-SCuBATenant' -Force
+
+InModuleScope Connection {
+    Describe -Tag 'Connection' -Name 'Disconnect-SCuBATenant' {
+        BeforeAll {
+            function Disconnect-MgGraph {}
+            Mock -ModuleName Connection Disconnect-MgGraph -MockWith {}
+            function Disconnect-ExchangeOnline {}
+            Mock -ModuleName Connection Disconnect-ExchangeOnline -MockWith {}
+            function Disconnect-SPOService {}
+            Mock -ModuleName Connection Disconnect-SPOService -MockWith {}
+            function Disconnect-PnPOnline {}
+            Mock -ModuleName Connection Disconnect-PnPOnline -MockWith {}
+            function Remove-PowerAppsAccount {}
+            Mock -ModuleName Connection Remove-PowerAppsAccount -MockWith {}
+            function Disconnect-MicrosoftTeams {}
+            Mock -ModuleName Connection Disconnect-MicrosoftTeams -MockWith {}
+            Mock -CommandName Write-Progress {}
+        }
+        It 'Disconnects from Microsoft Graph' {
+            {Disconnect-SCuBATenant -ProductNames 'aad'} | Should -Not -Throw
+        }
+        It 'Disconnects from Exchange Online' {
+            {Disconnect-SCuBATenant -ProductNames 'exo'} | Should -Not -Throw
+        }
+        It 'Disconnects from Defender (Exchange Online and Security & Compliance)' {
+            {Disconnect-SCuBATenant -ProductNames 'defender'} | Should -Not -Throw
+        }
+        It 'Disconnects from One Drive (SharePoint Online)' {
+            {Disconnect-SCuBATenant -ProductNames 'onedrive'} | Should -Not -Throw
+        }
+        It 'Disconnects from Power Platform' {
+            {Disconnect-SCuBATenant -ProductNames 'powerplatform'} | Should -Not -Throw
+        }
+        It 'Disconnects from SharePoint Online' {
+            {Disconnect-SCuBATenant -ProductNames 'sharepoint'} | Should -Not -Throw
+        }
+        It 'Disconnects from Microsoft Teams' {
+            {Disconnect-SCuBATenant -ProductNames 'sharepoint'} | Should -Not -Throw
+        }
+        It 'Disconnects from all products' {
+            {Disconnect-SCuBATenant} | Should -Not -Throw
+        }
```

Comment:
Wrote issue on calling disconnect multiple times for services

```suggestion
        It 'Disconnects from all products' {
            Disconnect-SCuBATenant
            Should -Invoke -CommandName Disconnect-MgGraph -Times 1 # Is called 3 times
            Should -Invoke -CommandName Disconnect-ExchangeOnline -Times 1 # Is called twice
            Should -Invoke -CommandName Disconnect-SPOService -Times 1 -Exactly
            Should -Invoke -CommandName Disconnect-PnPOnline -Times 1 -Exactly
            Should -Invoke -CommandName Remove-PowerAppsAccount -Times 1 -Exactly
            Should -Invoke -CommandName Disconnect-MicrosoftTeams -Times 1 -Exactly
        }
```

Reply:
not entirely sure what you mean by this. I think we can clear that up in the next meeting.